### PR TITLE
[KGA-97] Qa report 46

### PR DESCRIPTION
**Pull request by obatirou**
https://github.com/code-423n4/2024-09-kakarot-findings/issues/46

### CODE REVIEW

File: kakarot_scripts/deployment/evm_deployments.py [hunk 1]
```diff
@@ -64,6 +65,11 @@ async def deploy_evm_contracts():
         }
         await invoke("kakarot", "set_coinbase", int(contract.address, 16))
 
+    # %% Pre-fund precompiles
+    # see https://github.com/ethereum/go-ethereum/blob/5230b06d5151e214e80762eebed9196a670c52b1/core/vm/instructions.go#L404
+    for i in range(1, 10):
+        logger.info(f"ℹ️  YOOOOOOOOO {hex(i)}")
```
Comment (enitrat) [inline kakarot_scripts/deployment/evm_deployments.py:71]:
```suggestion
```

File: kakarot_scripts/deployment/evm_deployments.py [hunk 2]
```diff
@@ -64,6 +65,11 @@ async def deploy_evm_contracts():
...
+        await fund_address(hex(i), 1 / 1e18)
```
[the remark this answers is not in the document]
Comment (obatirou) [reply kakarot_scripts/deployment/evm_deployments.py:72]: We could assert on by getting balance and ensure is it > 0 indeed

File: kakarot_scripts/deployment/main.py [hunk 3]
```diff
@@ -92,6 +93,14 @@ async def main():
         f"💰  EVM deployer balance:\n    L2: {l2_balance} ETH\n    L1: {l1_balance} ETH"
     )
 
+    # check precompiles received funds
+    for i in range(1, 10):
```
Comment (enitrat) [inline kakarot_scripts/deployment/main.py:97]: I would not loop to 10 but make sure _all_ precompiles even non eth ones are warmed

can check my pending PR on warmup to see how the python vars have changed

**Review by enitrat — CHANGES_REQUESTED** [review]

File: kakarot_scripts/deployment/main.py (hunk 3 above)
Comment (obatirou) [reply kakarot_scripts/deployment/main.py]: Done

**Review by enitrat — APPROVED** [review]

**Pull request merged**